cpp/tests/io/parquet_chunked_reader_test.cu

```diff
@@ -1886,6 +1891,51 @@ TEST_F(ParquetChunkedReaderTest, TestNumRowsPerSourceEmptyTable)
     std::equal(expected_counts.cbegin(), expected_counts.cend(), num_rows_per_source.cbegin()));
 }
 
+TEST_F(ParquetReaderTest, ManyLargeLists)
+{
+  auto const stream = cudf::get_default_stream();
+
+  // Generate a list<bool> column with 10M outer rows and 2 bools per row
+  constexpr cudf::size_type num_rows      = 10'000'000;
+  constexpr cudf::size_type bools_per_row = 2;
+  auto offsets_iter = cudf::detail::make_counting_transform_iterator(0, offset_gen{bools_per_row});
+  auto offsets_col  = cudf::make_fixed_width_column(
+    cudf::data_type{cudf::type_id::INT32}, num_rows + 1, cudf::mask_state::UNALLOCATED);
+  thrust::copy(rmm::exec_policy_nosync(stream),
+               offsets_iter,
+               offsets_iter + num_rows + 1,
+               offsets_col->mutable_view().begin<int>());
+
+  auto bools_iter = cudf::detail::make_counting_transform_iterator(0, bool_gen{});
+  auto bools_col  = cudf::make_fixed_width_column(
+    cudf::data_type{cudf::type_id::BOOL8}, num_rows * bools_per_row, cudf::mask_state::UNALLOCATED);
+  thrust::copy(rmm::exec_policy_nosync(stream),
+               bools_iter,
+               bools_iter + (num_rows * bools_per_row),
+               bools_col->mutable_view().begin<bool>());
+
+  stream.synchronize();
+
+  auto list_col = cudf::make_lists_column(
+    num_rows, std::move(offsets_col), std::move(bools_col), 0, rmm::device_buffer{});
+
+  auto const table    = cudf::table_view({*list_col});
+  auto const filepath = temp_env->get_temp_filepath("ManyLargeLists.parquet");
+  auto const out_opts =
+    cudf::io::parquet_writer_options::builder(cudf::io::sink_info{filepath}, table).build();
+  // Write the table to parquet
+  cudf::io::write_parquet(out_opts);
+
+  // Concat filepath 150x for num rows to overflow cudf column size limits
+  constexpr cudf::size_type reads_to_overflow = 150;
```

Comment:
can we derive this? e.g.
```suggestion
  constexpr cudf::size_type reads_to_overflow = std::numeric_limits<cudf::size_type>::max() / (num_rows * bools_per_row) + 1;
```

Reply:
Nice suggestion. Applied

Reply:
hopefully this shaved off a few milliseconds